tests/test_cmdline.py

```diff
@@ -904,6 +904,30 @@ def test_cmdline_on_ignored_undeclared_option(
             ).splitlines(),
             id="generate_toml_config_with_a_setting_set_to_str_None",
         ),
+        pytest.param(
+            {"pyenvs": ["foo", "bar"]},
+            ["--pyenv", "baz", "xyzzy", "--generate-toml-config"],
+            dedent(
+                """\
+                # Copy this TOML section into your pyproject.toml to configure FawltyDeps
+                # (default values are commented)
+                [tool.fawltydeps]
+                # actions = ['check_undeclared', 'check_unused']
+                # output_format = 'human_summary'
+                # code = ['.']
+                # deps = ['.']
+                pyenvs = ['baz', 'xyzzy']
+                # custom_mapping_file = []
+                # ignore_undeclared = []
+                # ignore_unused = []
+                # deps_parser_choice = ...
+                # install_deps = false
+                # verbosity = 0
+                # [tool.fawltydeps.custom_mapping]
+                """
+            ).splitlines(),
+            id="generate_toml_config_with_multiple_pyenvs",
+        ),
```

Comment:
A general question on the code below. Apologies in advance as this hasn't been actually touched by this PR:
```python
    tmp_path = project_with_code_and_requirements_txt(
        imports=["requests"],
        declares=["pandas"],
    )
    setup_fawltydeps_config(config)
    output, errors, returncode = run_fawltydeps_subprocess(
        "--config-file=pyproject.toml", *args, cwd=tmp_path
    )
```
This seems to assume that the `tmp_path` returned by `project_with_code_and_requirements_txt()` and the one used by `setup_fawltydeps_config()` are the same. Is that the case? And if so, why?

Reply:
Yes, in the end all of this boils down to the [`tmp_path` fixture from pytest](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture): If I use multiple fixtures in a test, and each of those uses `tmp_path`, then it's all the _same_ `tmp_path`; it is instantiated _once per test_:

```python
@pytest.fixture
def fixtureA(tmp_path):
    return tmp_path

@pytest.fixture
def fixtureB(tmp_path):
    return tmp_path

def test_case(fixtureA, fixtureB):
    assert fixtureA == fixtureB
```

Both `project_with_code_and_requirements_txt()` and `setup_fawltydeps_config()` are fixtures that rely on the pytest `tmp_path` fixture (via their use of `write_tmp_files()`. And the path returned from `project_with_code_and_requirements_txt()` is the one returned from `write_tmp_files()`, which is indeed the `tmp_path` fixture/value itself.

Thus the `tmp_path` variable in the test below has the exact same value as if we had just used the `tmp_path` fixture directly in the test case.

The test case, as written, is a bit indirect and convoluted, for sure, but I don't think it's _misleading_?

This is maybe clearer, although a bit more verbose?

```python
    tmp_path = project_with_code_and_requirements_txt(
        imports=["requests"],
        declares=["pandas"],
    )
    config_path = setup_fawltydeps_config(config)
    output, errors, returncode = run_fawltydeps_subprocess(
        f"--config-file={config_path.relative_to(tmp_path)}",
        *args,
        cwd=tmp_path,
    )
```

Reply:
#TIL!
The new code you posted is as you said more verbose without necessarily making it clearer that we're using the same `tmp_path` as the fixture is called once.